### PR TITLE
Change CMAKE_CUDA_STANDARD to C++17 for Windows GPU build

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Otherwise it doesn't compile with the latest Visual Studio. Anyway, we will move to C++17 soon, so this change doesn't hurt. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/upsample.h [hunk 1]
```diff
@@ -18,8 +18,8 @@ constexpr const char* UpsampleModeCubic = "cubic";
 // is a 4x4 matrix
 const size_t CubicModeGridLength = 4;
 
-using GetNearestPixelFunc = std::function<int64_t(float, bool)>;
-using GetOriginalCoordinateFunc = std::function<float(float, float, float, float, float, float)>;
+using GetNearestPixelFunc = int64_t(*)(float, bool);
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/tensor/upsample.h:21]: Is this change required for c++17?

Comment (snnn) [reply onnxruntime/core/providers/cpu/tensor/upsample.h:21]: Yes, otherwise it doesn't build.

**Review by pranavsharma — APPROVED** [review]
